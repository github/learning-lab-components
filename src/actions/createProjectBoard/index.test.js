const createProjectBoard = require('./')
const mockContext = require('../../tests/mockContext')

describe('createProjectBoard', () => {
  let context

  beforeEach(() => {
    context = mockContext({}, {
      projects: {
        createRepoProject: jest.fn(() => Promise.resolve({ data: { id: 1 } })),
        createProjectColumn: jest.fn()
      }
    })
  })

  it('creates a new project board', async () => {
    await createProjectBoard(context, { name: 'My project board', description: 'Something clever' })
    expect(context.github.projects.createRepoProject).toHaveBeenCalled()
    expect(context.github.projects.createRepoProject.mock.calls).toMatchSnapshot()
  })

  it('creates a new project board with columns', async () => {
    await createProjectBoard(context, { name: 'My project board', description: 'Something clever', columns: ['To do', 'In progress', 'Done like dinner'] })
    expect(context.github.projects.createRepoProject).toHaveBeenCalled()
    expect(context.github.projects.createProjectColumn).toHaveBeenCalledTimes(3)
    expect(context.github.projects.createProjectColumn.mock.calls).toMatchSnapshot()
  })
})
