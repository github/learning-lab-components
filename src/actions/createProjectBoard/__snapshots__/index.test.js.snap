// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createProjectBoard creates a new project board 1`] = `
Array [
  Array [
    Object {
      "body": "Something clever",
      "name": "My project board",
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`createProjectBoard creates a new project board with columns 1`] = `
Array [
  Array [
    Object {
      "name": "To do",
      "project_id": 1,
    },
  ],
  Array [
    Object {
      "name": "In progress",
      "project_id": 1,
    },
  ],
  Array [
    Object {
      "name": "Done like dinner",
      "project_id": 1,
    },
  ],
]
`;
